packages/redux/src/entities/selectors/entity.ts

```diff
@@ -42,36 +40,27 @@ export const getEntities = <
 /**
  * Gets a specific entity of the given schema name, by id.
  *
- * @memberof module:entities/selectors
- *
- * @param {object} state - Application state.
- * @param {string} name - Entity/schema name.
- * @param {string|number} id - Entity identifier.
+ * @param state - Application state.
+ * @param name - Entity/schema name.
+ * @param id - Entity identifier.
  *
- * @returns {object|undefined} - The specific entity of the given id, undefined
+ * @returns - The specific entity of the given id, undefined
  * if it isn't found.
  */
-export const getEntityById = <
-  SchemaName extends keyof NonNullable<StateEntities>,
-  EntityId extends keyof NonNullable<NonNullable<StateEntities>[SchemaName]>,
->(
+export const getEntityById = <T>(
   state: StoreState,
   name: SchemaName,
   id: EntityId,
-): NonNullable<StateEntities>[SchemaName][EntityId] | undefined =>
-  getEntities(state, name)?.[id];
+): T => getEntities(state, name)?.[id];
```

Comment:
```suggestion
): T | undefined => getEntities(state, name)?.[id];
```

Not sure about this but, what about having the `undefined` part explicit here (because it's a reality since we have optional chaining) instead of having the caller to always add `| undefined`? Mainly because the caller can send what it wants, and if the undefined is missing, the types won't be correct.

For example here

```js
export const getAddress = (args): RandomType => getEntityById(args);
```

If I just set `RandomType`, it won't be true, because the result can actually be undefined as well... Does this make sense? 🤔

Reply:
If I explicitly set undefined as a possible response in the function, I have to explicitly set the same whenever I call the function. Because when I'm calling it, and I say that I'm only expecting lets say `RandomType` as a response, typescript gives an error saying that the function may also return undefined.

Reply:
And that's what's expected, right? I mean, if you just expect `RandomType` as a response, it's wrong, because it can actually be undefined as well, so the TS error makes sense, doesn't it? 🤔 _(If I understood correctly)_

Reply:
done 😁